airmozilla/starred/static/starred/js/star_event.js

```diff
@@ -70,6 +70,8 @@ var Stars = (function() {
                 // jquery array comparison http://stackoverflow.com/a/7726509/205832
                 if (!($(stars).not(serverStars).length === 0 &&
                       $(serverStars).not(stars).length === 0)) {
+                    stars = stars.concat(serverStars);
```

Comment:
This makes me nervous. What if `stars` is not empty or if there is an item in both arrays. Then you'd get repeated IDs in there. 
Why not do something like this (untested):
```javascript
serverStars.forEach(function(id) {
    if (stars.indexOf(id) === -1) { 
        stars.push(id); /// Should that be unshift() instead??!?
    }
});
```